### ISSUE TITLE
Error with "make pbccs"

### ISSUE THREAD

**Issue by a-velt** [PacificBiosciences/pitchfork#364]
Dear developers,

I tried to install pbccs, because it is asked by fc_quiver.py script of FALCON. I followed the instructions here : 
https://github.com/PacificBiosciences/unanimity/blob/master/doc/INSTALL.md

So, I type : 
> git clone https://github.com/PacificBiosciences/pitchfork
  cd pitchfork

But when I type : 
> make pbccs

I have the following errors : 
> Making all in tls
make[3]: Entering directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/tls'
  CC       tls.lo
  CC       tls_client.lo
  CC       tls_config.lo
  CC       tls_server.lo
  CC       tls_util.lo
  CC       tls_verify.lo
  CCLD     libtls.la
make[3]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/tls'
Making all in include
make[3]: Entering directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include'
Making all in openssl
make[4]: Entering directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include/openssl'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include/openssl'
make[4]: Entering directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include'
make[4]: Nothing to be done for `all-am'.
make[4]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include'
make[3]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/include'
Making all in apps
make[3]: Entering directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/apps'
  CC       apps.o
In file included from apps.c:138:0:
apps.h:283:8: error: redefinition of ‘struct option’
 struct option {
        ^
In file included from /usr/include/unistd.h:890:0,
                 from ../include/compat/unistd.h:10,
                 from apps.c:136:
/cm/shared/apps/gcc-5.3_install/gcc-5.3.0/include/getopt.h:82:8: note: originally defined here
 struct option
        ^
make[3]: *** [apps.o] Error 1
make[3]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/apps'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/cm/shared/apps/pitchfork/workspace/libressl-2.2.5'
make[1]: *** [/cm/shared/apps/pitchfork/workspace/libressl-2.2.5/ssl/.libs/libssl.so] Error 1
make[1]: Leaving directory `/cm/shared/apps/pitchfork/ports/thirdparty/libressl'
make: *** [openssl] Error 2

And I don't understand ... Have you an idea to help me ?

Best,
Amandine

**Comment by mhsieh** [comment]
Hi, Which linux distribution are you using?

**Comment by mhsieh** [comment]
This might have been an issue with the compiler environment you are using, let me see how I can advise to by-pass compiling python.

**Comment by mhsieh** [comment]
Scl6 is using python2.6, so you can just reuse system python to avoid build ssl. There are two options you can use,
1. fix your gcc-5.3_install/gcc-5.3.0 compiler or use compiler from SCL6 devtoolset-3 (might require system permission) instead of your gcc-5.3_install/gcc-5.3.0 compiler
2. find a python2.7 installation for SCL6 (might require system permission) or just use the python2.7 from an anaconda installation.

If you have root permission on this scl6 box, I can help you explore the use of SCL6 devtoolset-3. I can also help you on

**Comment by a-velt** [comment]
I used devtoolset-3 and all work fine, thank you very much for your help !! 
Just a notification, the command "pitchfork fetch --url" doesn't work for me, I had to replace all these commands by "wget" and all the installations work fine (blasr, pbalign, pbccs).

Best,
Amandine

**Comment by mhsieh** [comment]
Apparently you also need a newer curl in that case, I will add a fallback for older curl to wget.

**Comment by mhsieh** [comment]
I am closing this issue now. and the wget fallback issue is created https://github.com/PacificBiosciences/pitchfork/issues/367

**Comment by a-velt** [comment]
Thank you,

I have a last question about pitchfork. Because fc_quiver.py of Falcon requires ConsensusCore2, it is why I have installed pbccs. But when I launch the script, I have the error 

> "/cm/shared/apps/pitchfork/deployment/bin/variantCaller --algorithm=arrow -x 5 -X 120 -q 20 -j 24 -r /data2/avelt/Assembly_gewurz_161114/FALCON_GW_new-version/4-quiver/quiver_scatter
/001657F/001657F_ref.fa aln-001657F.bam -o /data2/avelt/Assembly_gewurz_161114/FALCON_GW_new-version/4-quiver/001657F/cns-001657F.fasta.gz -o /data2/avelt/Assembly_gewurz_161114/FALCO
N_GW_new-version/4-quiver/001657F/cns-001657F.fastq.gz
No module named ConsensusCore2
Traceback (most recent call last):
  File "/cm/shared/apps/FALCON-integrate/fc_env/lib/python2.7/site-packages/pbcommand/cli/core.py", line 136, in _pacbio_main_runner
    return_code = exe_main_func(*args, **kwargs)
  File "/cm/shared/apps/pitchfork/deployment/lib/python2.7/site-packages/GenomicConsensus-2.1.0-py2.7.egg/GenomicConsensus/main.py", line 381, in args_runner
    return tr.main()
  File "/cm/shared/apps/pitchfork/deployment/lib/python2.7/site-packages/GenomicConsensus-2.1.0-py2.7.egg/GenomicConsensus/main.py", line 308, in main
    self._algorithm = self._algorithmByName(options.algorithm, peekFile)
  File "/cm/shared/apps/pitchfork/deployment/lib/python2.7/site-packages/GenomicConsensus-2.1.0-py2.7.egg/GenomicConsensus/main.py", line 89, in _algorithmByName
    from GenomicConsensus.arrow import arrow
  File "/cm/shared/apps/pitchfork/deployment/lib/python2.7/site-packages/GenomicConsensus-2.1.0-py2.7.egg/GenomicConsensus/arrow/__init__.py", line 33, in <module>
    import utils
  File "/cm/shared/apps/pitchfork/deployment/lib/python2.7/site-packages/GenomicConsensus-2.1.0-py2.7.egg/GenomicConsensus/arrow/utils.py", line 41, in <module>
    import ConsensusCore2 as cc
ImportError: No module named ConsensusCore2
"

But I don't understand ... I follow instructions : 
https://github.com/PacificBiosciences/pitchfork/issues/310

I have successfully installed pbfalcon, pbreports, pbccs ... But I get the error. I have installed GenomicConsensus and ConsensusCore in the path /cm/shared/apps/pitchfork/deployment/ but I get the error again ... I don't know where is the problem ...

**Comment by mhsieh** [comment]
I will try and see if I can reproduce this on my scl6 box.

**Comment by mhsieh** [comment]
what is the output if you do this?
```
$ bash --init-file pitchfork/deployment/setup-env.sh 
bash-4.1$ python -c "import ConsensusCore2 as cc"
bash-4.1$ echo $?
0
bash-4.1$ 
```

**Comment by a-velt** [comment]
> bash --init-file /cm/shared/apps/pitchfork/deployment/setup-env.sh

> python -c "import ConsensusCore2 as cc"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named ConsensusCore2

> echo $?
1

Best,
Amandine

**Comment by mhsieh** [comment]
:(

**Comment by mhsieh** [comment]
My testing console log
[Terminal Saved Output.txt](https://github.com/PacificBiosciences/pitchfork/files/712510/Terminal.Saved.Output.txt)

**Comment by a-velt** [comment]
Hi,

I reinstalled pitchfork, getting the same outputs as you did, then launched the python command "python -c "import ConsensusCore2 as cc"" and ... it's ok, because I'm root during the installation and the test ! 
In fact, my folder containing the tools belongs to the root user and other users only have the right to run the tools but not write in the folder. When I launched fc_quiver.py, I did it with my user account ... and, if I understand well, python -c "import ConsensusCore2 as cc" try to write in the tool folder, and it's not possible.

Now, I launch fc_quiver.py as root and, I don't know why, it stops to run at one moment without specific error ... A colleague tell me that maybe it's due to my "maximum number of open files" parameter, he had this problem before, so, I test. But this is no longer your problem, pitchfork works fine.

Thank you again for your help and have a nice day,
Amandine
